store/src/java/com/zimbra/cs/store/StoreManager.java

```diff
@@ -18,45 +18,69 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
 
 import com.zimbra.common.localconfig.LC;
 import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.StringUtil;
 import com.zimbra.common.util.ZimbraLog;
 import com.zimbra.cs.account.Provisioning;
 import com.zimbra.cs.account.Server;
 import com.zimbra.cs.extension.ExtensionUtil;
 import com.zimbra.cs.imap.ImapDaemon;
 import com.zimbra.cs.mailbox.MailItem;
 import com.zimbra.cs.mailbox.Mailbox;
+import com.zimbra.cs.mailbox.util.MailItemHelper;
 import com.zimbra.cs.store.file.FileBlobStore;
+import com.zimbra.cs.store.helper.ClassHelper;
 import com.zimbra.cs.util.Zimbra;
+import com.zimbra.cs.volume.Volume;
+import com.zimbra.cs.volume.VolumeManager;
+import com.zimbra.soap.admin.type.StoreManagerRuntimeSwitchResult;
 
 public abstract class StoreManager {
 
+    // StoreManager readers cache
+    private static Map<Short, StoreManager> volumeStoreManagerCache = new ConcurrentHashMap<>();
     private static StoreManager sInstance;
+
+    private static String currentZimbraClassStore = LC.zimbra_class_store.value();
     private static Integer diskStreamingThreshold;
 
     public static StoreManager getInstance () {
-        if(ImapDaemon.isRunningImapInsideMailboxd()) {
-            return getInstance(LC.zimbra_class_store.value());
+        if (ImapDaemon.isRunningImapInsideMailboxd()) {
+            ZimbraLog.store.error("Actual StoreManager currentZimbraClassStore:%s, zimbra_class_store:%s", currentZimbraClassStore, LC.zimbra_class_store.value());
+            // if its different than current set in localConfig
+            synchronized (StoreManager.class) {
+                if (!LC.zimbra_class_store.value().equals(currentZimbraClassStore)) {
+                    ZimbraLog.store.error("currentZimbraClassStore:%s, is different than zimbra_class_store:%s", currentZimbraClassStore, LC.zimbra_class_store.value());
```

Comment:
Look around for more of these ERROR level logging statements.

Reply:
That was done while dev work, revisited and cleaned.